libphobos/testsuite/libphobos.cycles/cycles.exp

```diff
@@ -0,0 +1,62 @@
+# Copyright (C) 2017 Free Software Foundation, Inc.
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with GCC; see the file COPYING3.  If not see
+# <http://www.gnu.org/licenses/>.
+
+load_lib libphobos.exp
+load_lib libphobos-dg.exp
+
+global shouldfail
+set dg-output-text [list]
+
+# If a testcase doesn't have special options, use these.
+if ![info exists DEFAULT_DFLAGS] then {
+    set DEFAULT_DFLAGS "-g"
+}
+
+# Arguments to pass to the test program, expected output, and return code.
+set cycle_test_list [list \
+    { ignore ""  0 } \
+    { abort "object.Error@.*: Cyclic dependency between module mod. and mod." 1 } \
+    { print "Cyclic dependency between module mod3 and mod1" 0 } \
+    { deprecate "Deprecation 16211 warning:" 0 } \
```

Comment:
New library tests.  Compiles `mod*.d` in invokes the program with each given `--DRT-oncycle=` option.